src/main/java/org/manifold/compiler/front/Main.java

```diff
@@ -157,20 +296,25 @@ public Expression visitFunctionInvocationExpression(
     );
   }
 
+  // TODO KEY INSIGHT: combine the port type/port attributes and
+  // node attributes in a single FunctionTypeValue signature.
+  // As an example, if we have port types xIn(a: Int) and xOut(b: Int)
+  // and want a node type xDev whose attributes are p,q: Bool,
+  // input port u: xIn, output port v: xOut, we can declare it like
+  //
+  // xDev = primitive node (u: xIn, p: Bool, q: Bool) -> (v: xOut);
+  //
+  // and instantiate it like
+  //
+  // vResult = xDev(u: (0: uVal, 1: (a: 3)), p: True, q: False, v: (b: 4))
+  //
```

Comment:
What we're calling a `node` and what we're calling a `function` are really the same thing, semantically. :+1: This is partly why I don't like the graph theory terminology leaking into our domain objects.